PocketCastsTests/Tests/Common SwiftUI/List View Model/SearchableListViewModelTests.swift

```diff
@@ -0,0 +1,70 @@
+@testable import podcasts
+import XCTest
+
+@MainActor
+final class SearchableListViewModelTests: XCTestCase {
+    private var testItems: [TestableModel] = [
+        .init(title: "one"),
+        .init(title: "two"),
+        .init(title: "three"),
+        .init(title: "one two three")
+    ]
+
+    private var viewModel: SearchableListViewModel<TestableModel> = .init()
+
+    override func setUp() {
+        viewModel = SearchableListViewModel(items: testItems)
+    }
+
+    func testSearchingReturnsCorrectly() {
+        viewModel.search(with: "ONE")
+
+        XCTAssertEqual(viewModel.numberOfFilteredItems, 2)
+        XCTAssertEqual(viewModel.filteredItems, [testItems[0], testItems[3]])
+    }
+
+    func testSearchIgnoresPaddedSpaces() {
+        viewModel.search(with: "       one        ")
+
+        XCTAssertEqual(viewModel.numberOfFilteredItems, 2)
+        XCTAssertEqual(viewModel.filteredItems, [testItems[0], testItems[3]])
+    }
+
+    func testSearchResetsOnEmpty() {
+        viewModel.search(with: "ONE")
+
+        XCTAssertTrue(viewModel.isSearching)
+
+        viewModel.search(with: "")
+
+        XCTAssertFalse(viewModel.isSearching)
+    }
+
+    func testSearchResultsAreUpdatedWhenItemsChange() {
+        viewModel.search(with: "three")
+        viewModel.searchText = "three"
+
+        XCTAssertEqual(viewModel.numberOfFilteredItems, 2)
+
+        viewModel.items = testItems + [.init(title: "another three"), .init(title: "no")]
+
+        XCTAssertEqual(viewModel.numberOfFilteredItems, 3)
+    }
+
+    func testStopsSearchingWhenThereAreNoItems() {
+        viewModel.search(with: "three")
+        XCTAssertEqual(viewModel.numberOfFilteredItems, 2)
+
+        viewModel.items = []
+
+        XCTAssertEqual(viewModel.numberOfFilteredItems, 0)
+        XCTAssertFalse(viewModel.isSearching)
+    }
+
+    // MARK: - Test Model
+    private struct TestableModel: SearchableDataModel {
+        let title: String
+
+        var searchField: String { title }
```

Comment:
[nit] Maybe "searchableContent" is a more accurate name?

Reply:
Yeah, that works better. I updated here: de58d2fab4da1dfae3dd0359af75dd978b67b3fd